Change default rotation behaviour (#45)
CROCODILE-CESM/regional-mom6: docs/angle_calculation.md

Here we explain the implementation of MOM6 angle calculation in regional-mom6, which is the process by which regional-mom6 calculates the angle of curved horizontal grids (``hgrids``).

**Issue:** On a curved hgrid, we have to rotate the boundary conditions according to the angle the grid is rotated from lat-lon coordinates (true north vs model north). Although horizontal grids supplied by users will contain an `angle_dx` field, MOM6 ignores this field entirely and calculates its own grid angles internally. 

**Solution:** To be consistent with MOM6's treatement of grid angles, when we rotate our boundary conditions, we implemented MOM6 angle calculation in a file called "rotation.py", and included this in the the boundary regridding functions by default.

## Boundary rotation algorithm

> Steps 1-5 are the same as MOM6's internal angle calculation. Step 6 is an additional step required to apply this algorithm to the boundaries

Reviewer: ```suggestion
Steps 1-5 replicate the angle calculation as done by MOM6. Step 6 is an additional step required to apply this algorithm to the boundaries.
```

Reviewer: Either here, or in the docstring of the method that replicates the MOM6 angle calculation, it would be good to point to the MOM6 source code. I'd say the docstring is more appropriate place to do this, if it's not done already.

Reviewer: What does MOM6 do at the boundaries? Why don't we replicate that instead of using Step 6? Then we don't need to differentiate or explain, right?

    2. The "x" component is the same addition of differences in latitude.
    3. Thus, given the same units, we can call arctan to get the angle in degrees

6. **Additional step to apply to boundaries**

> Since the boundaries in a regional MOM6 model are on the q rather than t points, we need to expand the grid and calculate the angle at the boundary points. This is implemented in the `create_expanded_hgrid` function.

Reviewer: ```suggestion
Since the boundaries in a regional MOM6 model are on the `q` points and not on the `t` points, we need to expand the grid and calculate the angle at the boundary points. This is implemented in the `create_expanded_hgrid` function.
```

> ## Convert this method to boundary angles - 3 Options

Reviewer: ```suggestion
## Convert this method to boundary angles - 2 Options
```
